lib/graphql/query/arguments.rb

```diff
@@ -1,12 +1,51 @@
 # frozen_string_literal: true
 module GraphQL
   class Query
+    class StaticArguments
+      attr_reader :argument_definitions
+
+      def initialize(argument_definitions:)
+        @argument_definitions = argument_definitions
+      end
+
+      def instantiate_arguments(values)
+        arg_class.new(values, argument_definitions: argument_definitions)
+      end
+
+      private
+
+      def arg_class
+        @arg_class ||= begin
+          klass = Class.new(GraphQL::Query::Arguments).instance_exec(self) do |static_arguments|
+            static_arguments.argument_definitions.each do |_arg_name, arg_definition|
+              expose_as = arg_definition.expose_as.to_s
+
+              # Don't define a helper method if it would override something.
+              next if instance_methods.include?(expose_as)
```

Comment:
Wondering if a warning should be logged here? Otherwise it could be confusing.

Reply:
That seems helpful. What would be a good way to communicate that to the user? `warn(...)`?

Reply:
Not sure, this gem doesn't have a logger or even use `puts` very much for output like this /shrug

Reply:
@esjee @swalkinshaw would it make sense to warn or even raise an exception if we hit this condition? I think it will result in a nicer upgrade path when folks have to resolve an error instead of debugging weird behaviour caused by calling the original method.

Reply:
> debugging weird behaviour caused by calling the original method.

This also worries me. I would prefer defining methods by default, does that sound viable? 

Alternatively, we could: 

- `warn(...)` when we encounter a naming conflict, then: 
  - accept either `define_method:` which creates the method in case of a conflict OR 
  - `silence_warning:` which suppresses the warning in case of a conflict (to avoid log noise)

Reply:
> does that sound viable?

Unsure; I was running into trouble where some methods are defined _after_ this block, which would cause the methods to be overridden silently. Seems a bit flimsy. I'm unsure what a good solution is here; I'll add a `warn` for now.